frontend/src/store/enrolmentReport.js

```diff
@@ -0,0 +1,24 @@
+import { defineStore } from 'pinia';
+
+import EnrolmentReportService from '@/services/enrolmentReportService.js';
+import { useAppStore } from '@/store/app.js';
+
+export const useEnrolmentReport = defineStore('enrolmentReport', {
+  state: () => ({
+    hasDueReports: null,
+  }),
+  actions: {
+    async checkDueReports(organizationId, programYearId) {
+      const appStore = useAppStore();
+
+      const enrolmentReports = (
+        await Promise.all([
+          EnrolmentReportService.getEnrolmentReports(organizationId, programYearId),
+          EnrolmentReportService.getEnrolmentReports(organizationId, appStore.getPreviousProgramYearId(programYearId)),
+        ])
+      ).flat();
+
+      this.hasDueReports = enrolmentReports.some((report) => EnrolmentReportService.isDueEnrolmentReport(report));
```

Comment:
As discussed, this code retrieves all ERs for the current and previous programYear, even though we technically aren't using individual ERs. We only need to run some logic on them to determine the hasDueReports flag. To improve performance we can add a custom enrolmentReport endpoint to perform this logic for us and avoid the extra network traffic.

Reply:
Looks great!